test/modules/EdgeHubRestartTester/src/Program.cs

```diff
@@ -0,0 +1,318 @@
+
+                    // Wait until the specified restart period to do another restart
+                    await Task.Delay((int)(eachTestExpirationTime - DateTime.UtcNow).TotalMilliseconds, cts.Token);
+                }
+            }
+            catch (Exception e)
+            {
+                Logger.LogError($"Exception caught: {e}");
+                throw;
+            }
+            finally
+            {
+                iotHubServiceClient?.Dispose();
+                dmModuleClient?.Dispose();
+                msgModuleClient?.Dispose();
+            }
+
+            completed.Set();
+            handler.ForEach(h => GC.KeepAlive(h));
+            Logger.LogInformation("EdgeHubRestartTester Main() finished.");
+            return 0;
+        }
+
+        static TestResultBase CreateTestResult(
+            TestOperationResultType testOperationResultType,
+            DateTime edgeHubRestartedTime,
+            HttpStatusCode edgeHubRestartStatusCode,
+            DateTime completedTime,
+            HttpStatusCode completedStatus,
+            Guid batchId,
+            uint restartSequenceNumber,
+            long sequenceNumber)
+        {
+            switch (testOperationResultType)
+            {
+                case TestOperationResultType.Messages:
+                    return new EdgeHubRestartMessageResult(
+                        Settings.Current.ModuleId + testOperationResultType.ToString(),
+                        DateTime.UtcNow,
+                        Settings.Current.TrackingId,
+                        batchId.ToString(),
+                        sequenceNumber.ToString(),
+                        edgeHubRestartedTime,
+                        edgeHubRestartStatusCode,
+                        completedTime,
+                        completedStatus,
+                        restartSequenceNumber);
+
+                case TestOperationResultType.DirectMethod:
+                    return new EdgeHubRestartDirectMethodResult(
+                        Settings.Current.ModuleId + testOperationResultType.ToString(),
+                        DateTime.UtcNow,
+                        Settings.Current.TrackingId,
+                        batchId,
+                        sequenceNumber.ToString(),
+                        edgeHubRestartedTime,
+                        edgeHubRestartStatusCode,
+                        completedTime,
+                        completedStatus,
+                        restartSequenceNumber);
+
+                default:
+                    throw new NotSupportedException($"{testOperationResultType} is not supported in CreateEdgeHubRestartTestResult()");
+            }
+        }
+
+        static async Task<Tuple<DateTime, HttpStatusCode>> SendDirectMethodAsync(
+            string deviceId,
+            string targetModuleId,
+            ModuleClient moduleClient,
+            string directMethodName,
+            DateTime testExpirationTime,
```

Comment:
rename to runExpirationTime or others, but not testExpirationTime which is already used in main program.

Reply:
`runExpirationTime`